pass.go

```diff
@@ -0,0 +1,1054 @@
+	}
+
+	fmt.Printf("Loading config file \"%s\"\n", configFileName)
+
+	decoder := json.NewDecoder(file)
+	if err = decoder.Decode(&cfg); err != nil {
+		return cfg, fmt.Errorf("Unable to decode config file: %v", err)
+	}
+
+	if len(cfg.Secret) < 64 {
+		return cfg, fmt.Errorf("Site secret shorter than 64 characters, please make it at least 64 characters")
+	}
+
+	fmt.Printf("Config file \"%s\" successfully loaded\n----\n", configFileName)
+	return cfg, nil
+}
+
+// recoverWrap registers a deferred error handler for the http.Handler
+// This prevents the entire program from coming down like a ton of bricks if a
+// user causes a panic.
+// The stacktrace is printed and the corresponding handler dies, but no other
+// requests are affected.
+func recoverWrap(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var err error
+		defer func() {
+			r := recover()
+			if r != nil {
+				switch t := r.(type) {
+				case string:
+					err = errors.New(t)
+				case error:
+					err = t
+				default:
+					err = fmt.Errorf("recoverWrap Unknown error %T %v", r, r)
+				}
+
+				log.Printf("recoverWrap triggered!\n")
+				fmt.Printf("Cause: %v\n|\n", err)
+
+				msgStrings := strings.Split(strings.TrimRight(string(debug.Stack()), "\n"), "\n")
+
+				// 7 is to remove the heading and the first 3 frames (2 lines each)
+				// because they contain the call to the recoverWrap func and the
+				// call to panic() which is useless when _this_ program is the issue
+				msgStrings = msgStrings[7:]
+
+				for _, value := range msgStrings {
+					fmt.Printf("|\t%s\n", string(value))
+				}
+
+				fmt.Printf("|\nEND of wrapped panic\n")
+
+				// We paniced, assume templating is a no-go and just output plaintext error
+				http.Error(w, "Server error", http.StatusInternalServerError)
+			}
+		}()
+
+		h.ServeHTTP(w, r)
+	})
+}
+
+// RequestHandler is the interface an importer must implement to hook into the
+// request handler.
+type RequestHandler interface {
+	Handle(hi http.Handler) (ho http.Handler)
+}
+
+// RegisterRequestHandler allows an importer to hook the request handler, allows
+// a sort of MITM for incoming requests, RequestHandlers must call f with a Pass
+// instance if they want the request to continue via the main program logic
+func (p *Pass) RegisterRequestHandler(f func(http.Handler) http.Handler) {
+	_, filename, line, _ := runtime.Caller(1)
+	fmt.Printf("Added RequestHandler %s:%d\n----\n", filename, line)
+	p.handler = f(p.handler)
+}
+
+// Pass is a struct holding all required variables for wrappers, instanced using
+// the CreatePass function
+type Pass struct {
+	sv           *serverVariables
+	baseTemplate *template.Template
+	handler      http.Handler
+}
+
+// CreatePass returns a pointer to an instance of Pass
+// for use in calls to Main(Pass) in main() functions of hooking applications
+func CreatePass() *Pass {
+	log.SetFlags(0)
```

Comment:
Don't do this here. Setting the log flag in `main` alone is sufficient.

Reply:
Fixed in next push